sam/feature_engineering/simple_feature_engineering.py

```diff
@@ -0,0 +1,198 @@
+        For supported component values, see `SimpleFeatureEngineer.valid_components`
+        (e.g. "second_of_day", "hour_of_day). Valid types are "onehot" or "cyclical".
+    time_col : str (default=None)
+        Name of the time column (e.g. "TIME"). If None, the index of the dataframe is used.
+    timezone: str, optional (default=None)
+        if tz is not None, convert the time to the specified timezone, before creating features.
+        timezone can be any string that is recognized by pytz, for example `Europe/Amsterdam`.
+        We assume that the TIME column is always in UTC,
+        even if the datetime object has no tz info.
+    drop_first : bool (default=True)
+        Whether to drop the first value of time components (used for onehot encoding)
+    keep_original : bool (default=False)
+        Whether to keep the original columns in the dataframe.
+
+    Example
+    -------
+    >>> from sam.feature_engineering import SimpleFeatureEngineer
+    >>> from sam.data_source import read_knmi
+    >>> data = read_knmi('2018-02-01', '2019-10-01', latitude=52.11, longitude=5.18, freq='hourly',
+    ...                  variables=['FH', 'FF', 'FX', 'T', 'TD', 'SQ', 'Q', 'DR', 'RH', 'P',
+    ...                             'VV', 'N', 'U', 'IX', 'M', 'R', 'S', 'O', 'Y'])
+    >>> y = data['T'].shift(-1)
+    >>> X = data.drop('T', axis=1)
+    >>> fe = SimpleFeatureEngineer(rolling_features=[('T', 'mean', 12), ('T', 'mean', '24')],
+    ...                            time_features=[('hour_of_week', 'onehot')])
+    >>> X_fe = fe.fit_transform(X, y)
+    >>> X_fe.head()
+    """
+
+    component_range = {
+        "second_of_minute": (0, 59),
+        "second_of_hour": (0, 3599),
+        "second_of_day": (0, 86399),
+        "minute_of_hour": (0, 59),
+        "minute_of_day": (0, 1439),
+        "hour_of_day": (0, 23),
+        "hour_of_week": (0, 167),
+        "day_of_week": (1, 7),
+        "day_of_month": (1, 31),
+        "day_of_year": (1, 366),
+        "week_of_year": (1, 53),
+        "month_of_year": (1, 12),
+    }
+
+    component_function = {
+        "second_of_minute": lambda x: x.dt.second,
+        "second_of_hour": lambda x: x.dt.second + x.dt.minute * 60,
+        "second_of_day": lambda x: x.dt.second + x.dt.minute * 60 + x.dt.hour * 3600,
+        "minute_of_hour": lambda x: x.dt.minute,
+        "minute_of_day": lambda x: x.dt.minute + x.dt.hour * 60,
+        "hour_of_day": lambda x: x.dt.hour,
+        "hour_of_week": lambda x: x.dt.hour + x.dt.weekday * 24,
+        "day_of_week": lambda x: x.dt.isocalendar().day,
+        "day_of_month": lambda x: x.dt.day,
+        "day_of_year": lambda x: x.dt.dayofyear,
+        "week_of_year": lambda x: x.dt.isocalendar().week,
+        "month_of_year": lambda x: x.dt.month,
+    }
+
+    valid_components = component_range.keys()
+
+    def __init__(
+        self,
+        rolling_features: Optional[Union[List[Tuple], pd.DataFrame]] = [],
```

Comment:
I wouldn't use `[]` as default parameters, since lists are mutable, which is always risky. I would suggest either using tuples or None as default